gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/sampling/PageRankSamplingTest.java

```diff
@@ -68,89 +64,68 @@ public SamplingAlgorithm getSamplingOperator() {
-              assertTrue("sampled vertex has PageRankScore smaller or equal than threshold",
-                score > sampleSize);
-            } else {
-              assertTrue("sampled vertex has PageRankScore greater than threshold",
-                score <= sampleSize);
-            }
-          }
-        }
+    for (Vertex v : newVertices) {
+      if (v.hasProperty(SamplingConstants.SCALED_PAGE_RANK_SCORE_PROPERTY_KEY)) {
+        noneHasScore = false;
       } else {
-        if (keepVerticesIfSameScore) {
-          assertEquals("not all vertices got sampled (should be, all got same score)",
-            dbVertices.size(), newVertices.size());
-        } else {
-          assertTrue("some vertices got sampled (should NOT be, all got same score)",
-            newVertices.isEmpty());
-        }
+        allHaveScore = false;
       }
-      dbEdges.removeAll(newEdges);
-      for (Edge edge : dbEdges) {
-        assertFalse("edge from original graph was not sampled but source and target were",
-          newVertexIDs.contains(edge.getSourceId()) &&
-            newVertexIDs.contains(edge.getTargetId()));
-      }
-
-      // test special graph
-      LogicalGraph specialGraph = getLoaderFromString(
-        "(alice:Person {name : \"Alice\"})\n" +
-          "(eve:Person {name : \"Eve\"})\n" +
-          "g0:Community {interest : \"Friends\", vertexCount : 2} [\n" +
-          "(eve)-[eka:knows {since : 2013}]->(alice)\n" +
-          "(alice)-[ake:knows {since : 2013}]->(eve)]").getLogicalGraphByVariable("g0");
-
-      LogicalGraph sampledGraph = getSamplingOperator().sample(specialGraph);
-
-      List<Vertex> specialVertices = Lists.newArrayList();
-      List<Vertex> specialSampledVertices = Lists.newArrayList();
-
-      specialGraph.getVertices().output(new LocalCollectionOutputFormat<>(specialVertices));
-      sampledGraph.getVertices().output(new LocalCollectionOutputFormat<>(specialSampledVertices));
-
-      getExecutionEnvironment().execute();
-
-      assertNotNull("graph was null", sampledGraph);
-
-      GraphHead specialGh = sampledGraph.getGraphHead().collect().get(0);
-      double minScore1 = specialGh.getPropertyValue(
-        SamplingConstants.MIN_PAGE_RANK_SCORE_PROPERTY_KEY).getDouble();
-      double maxScore1 = specialGh.getPropertyValue(
-        SamplingConstants.MAX_PAGE_RANK_SCORE_PROPERTY_KEY).getDouble();
-
-      assertEquals("min PageRankScore is not equal to max PageRankScore",
-        minScore1, maxScore1, 0.0);
+    }
 
-      if (keepVerticesIfSameScore) {
-        assertEquals(
-          "special: not all vertices got sampled (should be, all got same score)",
-          specialVertices.size(), specialSampledVertices.size());
-      } else {
-        assertTrue(
-          "special: some vertices got sampled (should NOT be, all got same score)",
-          specialSampledVertices.isEmpty());
+    assertTrue("some vertices do and some do not have scaled PageRank-score property",
```

Comment:
Please add some inline comments next time if the logic gets more complicated. It's kinda hard to understand the conditions and assertions without them